bootstrap/v2/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -577,84 +572,392 @@ func (kustomize *kustomize) writeConfigFile() error {
-	cfgFilePathErr := ioutil.WriteFile(cfgFilePath, buf, 0644)
-	if cfgFilePathErr != nil {
-		return cfgFilePathErr
+	updateConfigMapArgs := func(parentConfigMapArgs *types.ConfigMapArgs, childConfigMapArgs types.ConfigMapArgs) {
+		parentConfigMapArgs.Name = childConfigMapArgs.Name
+		parentConfigMapArgs.Namespace = childConfigMapArgs.Namespace
+		if childConfigMapArgs.EnvSource != "" {
+			envAbsolutePathSource := path.Join(targetDir, childConfigMapArgs.EnvSource)
+			envSource := extractSuffix(compDir, envAbsolutePathSource)
+			parentConfigMapArgs.EnvSource = envSource
+		}
+		if childConfigMapArgs.FileSources != nil && len(childConfigMapArgs.FileSources) > 0 {
+			parentConfigMapArgs.FileSources = make([]string,0)
+			for _, fileSource := range childConfigMapArgs.FileSources {
+				fileAbsolutePathSource := path.Join(targetDir, fileSource)
+				parentConfigMapArgs.EnvSource = extractSuffix(compDir, fileAbsolutePathSource)
+			}
+		}
+		if childConfigMapArgs.LiteralSources != nil && len(childConfigMapArgs.LiteralSources) > 0 {
+			parentConfigMapArgs.LiteralSources = make([]string,0)
+			for _, literalSource := range childConfigMapArgs.LiteralSources {
+				parentConfigMapArgs.LiteralSources = append(parentConfigMapArgs.LiteralSources, literalSource)
+			}
+		}
+		switch types.NewGenerationBehavior(childConfigMapArgs.Behavior) {
+		case types.BehaviorCreate:
+			if _, ok := kustomizationMaps[configMapGeneratorMap][childConfigMapArgs.Name]; !ok {
+				parent.ConfigMapGenerator = append(parent.ConfigMapGenerator, *parentConfigMapArgs)
+				kustomizationMaps[configMapGeneratorMap][childConfigMapArgs.Name] = true
+			}
+		case types.BehaviorMerge, types.BehaviorReplace, types.BehaviorUnspecified:
+			fallthrough
+		default:
+			parent.ConfigMapGenerator = append(parent.ConfigMapGenerator, *parentConfigMapArgs)
+			kustomizationMaps[configMapGeneratorMap][childConfigMapArgs.Name] = true
+		}
+	}
+
+	updateParamFiles()
+	if child.Bases == nil {
+		basePath := extractSuffix(compDir, targetDir)
+		if _, ok := kustomizationMaps[basesMap][basePath]; !ok {
+			parent.Bases = append(parent.Bases, basePath)
+			kustomizationMaps[basesMap][basePath] = true
+		}
+		return nil
+	}
+	for _, value := range child.Bases {
+		baseAbsolutePath := path.Join(targetDir, value)
+		basePath := extractSuffix(compDir, baseAbsolutePath)
+		if _, ok := kustomizationMaps[basesMap][basePath]; !ok {
+			parent.Bases = append(parent.Bases, basePath)
+			kustomizationMaps[basesMap][basePath] = true
+		} else {
+			childPath := extractSuffix(compDir, targetDir)
+			kustomizationMaps[basesMap][childPath] = true
+		}
+	}
+	if child.NamePrefix != "" {
+		log.Warnf("cannot merge nameprefix %v ", child.NamePrefix)
+	}
+	if child.GeneratorOptions != nil && parent.GeneratorOptions == nil {
+		parent.GeneratorOptions = child.GeneratorOptions
+	}
+	if child.NameSuffix != "" {
+		log.Warnf("cannot merge namesuffix %v ", child.NamePrefix)
+
+	}
+	if len(child.CommonLabels) > 0 {
```

Comment:
ignore all CommonLabels and others? saying I have no overlay, only `base` as line 826, should we still drop these?
maybe for these, we don't try to merge, just use the first one, that is 
```
if len(child.CommonAnnotations) > 0 &&  len(parent.CommonAnnotations) == 0 {
   parent.CommonAnnotations = child.CommonAnnotations  // maybe push them one by one
}
```

Reply:
sorry, I just didn't find the code to include `base`